src/LessCompiler.ts

```diff
@@ -67,6 +68,11 @@ export async function compile(
 
   options.plugins.push(new LessDocumentResolverPlugin());
 
+  // If options.rootFileInfo is not undefined, less will use the filepath to read content again
+  if (useContent) {
+    delete options.rootFileInfo;
+  }
+
```

Comment:
If we assume the above, then I think this can be replaced with:

```typescript
   if (preprocessors.length > 0) {
     content = preprocessors.reduce((input, p) => p(input), content);
   }
```

Although a plain loop might be even clearer than a reducer here.